velox/expression/CastExpr.cpp

```diff
@@ -110,19 +110,20 @@ void applyDecimalCastKernel(
   });
 }
 
-template <typename TOutput>
-void applyBigintToDecimalCastKernel(
+template <TypeKind InputKind, typename TOutput>
+void applyIntToDecimalCastKernel(
     const SelectivityVector& rows,
     const BaseVector& input,
     exec::EvalCtx& context,
     const TypePtr& toType,
     VectorPtr castResult) {
-  auto sourceVector = input.as<SimpleVector<int64_t>>();
+  using TInput = typename TypeTraits<InputKind>::NativeType;
+  auto sourceVector = input.as<SimpleVector<TInput>>();
```

Comment:
Can we figure this from the BaseVector input ? Does this necessarily have to be a type template variable ?

Reply:
You are still introducing a runtime check with the case statement below.